controllers/databaseclusterbackup_controller.go

```diff
@@ -226,12 +414,18 @@ func (r *DatabaseClusterBackupReconciler) reconcilePXC(ctx context.Context, back
 	}
 	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, pxcCR, func() error {
 		pxcCR.TypeMeta = metav1.TypeMeta{
-			APIVersion: pxcBackupAPI,
+			APIVersion: pxcAPIVersion,
 			Kind:       pxcBackupKind,
 		}
 		pxcCR.Spec.PXCCluster = backup.Spec.DBClusterName
 		pxcCR.Spec.StorageName = backup.Spec.BackupStorageName
 
+		pxcCR.SetOwnerReferences([]metav1.OwnerReference{{
```

Comment:
I recommend checking how `BlockOwnerDeletion` is set. Kubernetes says it's `true` by default while `apimachinery` says it's `false`. It's better to check it is configured as we need it.

Reply:
@oksana-grishchenko are you sure that this is actually needed?
It's my understanding that calling `controllerutil.SetControllerReference(backup, pxcCR, r.Client.Scheme())` in line 412 already does this.
AFAIK `SetControllerReference` is basically a `SetOwnerReferences` with the controller flag set to true, so I believe that this section of code actually does nothing.

Reply:
It's the second paragraph on this page https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications

Reply:
Checked the `controllerutil.SetControllerReference` function - sets the `BlockOwnerDeletion` to `true` under the hood, so I added it too. 

@recharte I decided to get rid of the `controllerutil.SetControllerReference` in this controller, bc it used to work well in case we _create_ the upstream backup, but it didn't work well when the upstream backup `was already there`. To avoid two updates on a single object during one method, I decided to use `SetOwnerReference()` instead and have full control over the update.

Reply:
Out of curiosity, what do you mean with "didn't work well"?

I'm guessing that it might error out because if the DB operator had already set the controller flag in the owner reference than we wouldn't be able to do the same.
> Since only one OwnerReference can be a controller, it returns an error if there is another OwnerReference with Controller flag set.
https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetControllerReference

Was this the case?

Reply:
@recharte no, the problem appeared in case the upstream resource already exists and it only needs to update its OwnerReference (it happens when a scheduled backup is performed). 
The `controllerutil.SetControllerReference` din't take any effect, because everything that happens in the later `controllerutil.CreateOrUpdate` doesn't change the CR, so the function returns "unchanged" and the OwnerReference got never updated.

Reply:
Makes sense :slightly_smiling_face: thanks for the explaination.